OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -397,42 +456,70 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 						prefix += "|";
 				}
 
-				var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), Game.ModData.VideoLoaders);
-				if (video != null)
+				var fileExtension = Path.GetExtension(filename.ToLowerInvariant());
+				if (allowedSpriteExtensions.Contains(fileExtension))
 				{
-					player = panel.Get<VideoPlayerWidget>("PLAYER");
-					player.Load(prefix + filename);
-					player.DrawOverlay = false;
-					isVideoLoaded = true;
+					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
+					currentFrame = 0;
 
 					if (frameSlider != null)
 					{
-						frameSlider.MaximumValue = (float)player.Video.Frames - 1;
-						frameSlider.Ticks = 0;
+						frameSlider.MaximumValue = (float)currentSprites.Length - 1;
+						frameSlider.Ticks = currentSprites.Length;
 					}
 
-					return true;
+					currentVoxel = null;
 				}
-
-				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vxl")
+				else if (allowedModelExtensions.Contains(fileExtension))
 				{
 					var voxelName = Path.GetFileNameWithoutExtension(filename);
 					currentVoxel = world.ModelCache.GetModel(voxelName);
 					currentSprites = null;
 				}
-				else
+				else if (allowedAudioExtensions.Contains(fileExtension))
 				{
-					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
-					currentFrame = 0;
+					// Mute music so it doesn't interfere with the current asset.
+					MuteSounds();
 
-					if (frameSlider != null)
+					using (var soundStream = Game.ModData.DefaultFileSystem.Open(prefix + filename))
 					{
-						frameSlider.MaximumValue = (float)currentSprites.Length - 1;
-						frameSlider.Ticks = currentSprites.Length;
+						foreach (var modDataSoundLoader in Game.ModData.SoundLoaders)
+						{
+							if (modDataSoundLoader.TryParseSound(soundStream, out currentSoundFormat))
```

Comment:
It's somewhat cheeky to rely on `currentSoundFormat` having working methods after being disposed here, however I'm willing to live with it.

Reply:
I put a bunch of effort into making the cheekiness work for AUD and WAV, but apparently it still breaks on OGG and MP3 (no idea about VOC atm), so will need to find another way I guess.

Reply:
Ultimately brought back `currentAudioStream` as to not have to dispose the stream prematurely. Hopefully nothing bad comes out of that.